packages/amplify-util-mock/src/api/lambda-invoke.ts

```diff
@@ -1,39 +1,98 @@
-import { $TSAny, $TSContext } from "amplify-cli-core";
+import { $TSAny, $TSContext, AmplifyFault, AMPLIFY_SUPPORT_DOCS } from "amplify-cli-core";
 import { loadLambdaConfig } from '../utils/lambda/load-lambda-config';
-import { BuildType } from 'amplify-function-plugin-interface';
+import { BuildType, FunctionRuntimeLifecycleManager, BuildRequest } from 'amplify-function-plugin-interface';
 import { getInvoker, getBuilder } from 'amplify-category-function';
 import { timeConstrainedInvoker } from '../func';
 import { printer } from 'amplify-prompts';
+import { LambdaTrigger, LambdaTriggerConfig } from "../utils/lambda/find-lambda-triggers";
 
 /**
  * Utility method to invoke the lambda function locally. 
  * Ensures latest function changes are built before invoking it.
  * @param context The CLI context
- * @param functionName Lambda function to invoke locally
+ * @param trigger Lambda trigger to invoke locally
  * @param data Data to be passed to the local lambda function invocation.
  */
-export const invokeLambda = async (context: $TSContext, functionName: string, data: $TSAny): Promise<void> => {
+export const invokeTrigger = async (context: $TSContext, trigger: LambdaTrigger, data: $TSAny): Promise<void> => {
+  let invoker: $TSAny;
```

Comment:
A more accurate type should be possible, ideally not using `$TSAny`

Something like:
```ts
  let invoker: ({ event: $TSAny }) => Promise<$TSAny>;
```